sdks/java/core/src/test/java/org/apache/beam/sdk/options/ProxyInvocationHandlerTest.java

```diff
@@ -683,10 +722,233 @@ public void testJsonConversionOfSerializableWithMetadataProperty() throws Except
     assertEquals("TestString", options2.getValue().getValue());
   }
 
+  @Test
+  public void testDisplayDataItemProperties() {
+    PipelineOptions options = PipelineOptionsFactory.create();
+    options.setTempLocation("myTemp");
+    DisplayData displayData = DisplayData.from(options);
+
+    assertThat(displayData, hasDisplayItem(allOf(
+        hasKey("tempLocation"),
+        hasType(DisplayData.Type.STRING),
+        hasValue("myTemp"),
+        hasNamespace(PipelineOptions.class)
+    )));
+  }
+
+  @Test
+  public void testDisplayDataTypes() {
+    Instant now = Instant.now();
+
+    TypedOptions options = PipelineOptionsFactory.as(TypedOptions.class);
+    options.setInteger(1234);
+    options.setTimestamp(now);
+    options.setJavaClass(ProxyInvocationHandlerTest.class);
+    options.setObject(new Serializable() {
```

Comment:
All options need to be json serializable or marked with @Hidden

It seems as though for PipelineOptions, you could just use the JSON version of the data if you don't understand the type instead of doing toString().

This is just a comment as to whether you considered just using the json representation of each option instead of the toString version.